src/mongocxx/test/collection_mocked.cpp

```diff
@@ -55,6 +56,31 @@ TEST_CASE("Collection", "[collection]") {
     collection mongo_coll = mongo_db[collection_name];
     REQUIRE(mongo_coll);
 
+    SECTION("Collection Rename", "[collection]") {
```

Comment:
As long as you're adding this test, could you also add an integration test for rename?